server/src/pages/shot/text-tool.js

```diff
@@ -73,10 +76,16 @@ exports.TextTool = class TextTool extends React.Component {
     }
   }
 
-  adjustHeight() {
+  adjustHeightAndWidth() {
     const styles = window.getComputedStyle(this.textInput.current);
     const height = Math.ceil(parseFloat(styles.lineHeight) + parseFloat(styles.paddingTop) + parseFloat(styles.paddingBottom));
     this.textInput.current.style.minHeight = `${height}px`;
+
+    inputTextSize = parseFloat(styles.fontSize);
+    inputTextPadding = parseFloat(styles.paddingLeft);
+
+    const width = Math.ceil(((inputTextSize / 2) * this.textInput.current.value.length) + parseFloat(styles.paddingLeft) + parseFloat(styles.paddingRight));
```

Comment:
Assumption here single character width is fontSize / 2 is flawed resulting in incorrect x position on hit of confirm